proteuslib/util/tests/test_initialization.py

```diff
@@ -27,32 +29,97 @@
+        with pytest.raises(ValueError, match=msg):
+            check_dof(m, fail_flag=True)
+        with pytest.raises(ValueError, match=msg):
+            assert_no_degrees_of_freedom(m)
+
+    @pytest.mark.unit
+    def test_zero(self, m):
+        m.a.fix(5)
+        # check should pass since DOF=0
         check_dof(m, fail_flag=True)
-    m.a.fix(5)
-    # check should pass since DOF=0
-    check_dof(m, fail_flag=True)
-
-@pytest.mark.unit
-def test_check_solve():
-    m = ConcreteModel()
-    m.a = Var()
-    m.acon = Constraint(rule=m.a >= 10)
-    m.bcon = Constraint(rule=m.a == 5)
-    results = solver.solve(m)
-    # check_solve should pass since fail_flag=False and only warning will be produced
-    check_solve(results, logger=_log, fail_flag=False)
-    # expect the solve to fail and raise error
-    with pytest.raises(ValueError, match="The solver failed to converge to an optimal solution. This suggests that the "
-                                         "user provided infeasible inputs or that the model is poorly scaled."):
-        check_solve(results, logger=_log, fail_flag=True)
\ No newline at end of file
+        check_dof(m, fail_flag=True, expected_dof=0)
+        assert_no_degrees_of_freedom(m)
+
+        m.a.unfix()
+
+
+class TestCheckSolve:
+    @pytest.fixture(scope="class")
+    def m(self):
+        m = ConcreteModel()
+        m.a = Var()
+        m.acon = Constraint(rule=m.a >= 10)
+        m.bcon = Constraint(rule=m.a == 5)
+        return m
+
+    @pytest.mark.unit
+    def test_failure(self, m):
+        results = solver.solve(m)
+        # check_solve should pass since fail_flag=False and only warning will be produced
+        check_solve(results, logger=_log, fail_flag=False)
+        # expect the solve to fail and raise error
+        with pytest.raises(ValueError, match="The solver failed to converge to an optimal solution. This suggests that the "
+                                             "user provided infeasible inputs or that the model is poorly scaled."):
+            check_solve(results, logger=_log, fail_flag=True)
+
+    @pytest.mark.unit
+    def test_failure_checkpoint(self, m):
+        results = solver.solve(m)
+        # check_solve should pass since fail_flag=False and only warning will be produced
+        check_solve(results, checkpoint='test', logger=_log, fail_flag=False)
+        # expect the solve to fail and raise error
+        with pytest.raises(ValueError, match="test failed. The solver failed to converge to an optimal solution. "
+                "This suggests that the user provided infeasible inputs or that the model is poorly scaled."):
+            check_solve(results, checkpoint='test', logger=_log, fail_flag=True)
+    
+    @pytest.mark.unit
+    def test_success(self, m):
+        m.acon.deactivate()
+
+        results = solver.solve(m)
+        # check_solve should pass since fail_flag=False and only warning will be produced
```

Comment:
Does this comment still stand or was it left from before?

Reply:
Good catch -- incorrect comments are worse than no comment!